h2o-core/src/main/java/hex/Model.java

```diff
@@ -1859,12 +1861,44 @@ protected Frame predictScoreImpl(Frame fr, Frame adaptFrm, String destination_ke
                                    j,
                                    customMetricFunc).doAll(names.length, Vec.T_NUM, adaptFrm);
 
-    if (computeMetrics && bs._mb != null) //metric builder can be null if training was interrupted/cancelled
-      bs._mb.makeModelMetrics(this, fr, adaptFrm, bs.outputFrame());
-    Frame predictFr = bs.outputFrame(Key.<Frame>make(destination_key), names, domains);
-    return postProcessPredictions(adaptFrm, predictFr, j);
+    ModelMetrics.MetricBuilder<?> mb = null;
+    Frame rawPreds = null;
+    if (computeMetrics && bs._mb != null) {
+      rawPreds = bs.outputFrame();
+      mb = bs._mb;
+    }
+    Frame predictFr = bs.outputFrame(Key.make(destination_key), names, domains);
+    Frame outputPreds = postProcessPredictions(adaptFrm, predictFr, j);
+    return new PredictScoreResult(mb, rawPreds, outputPreds);
   }
 
+  protected class PredictScoreResult {
+    private final ModelMetrics.MetricBuilder<?> _mb; // metric builder can be null if training was interrupted/cancelled even when metrics were requested
+    private final Frame _rawPreds;
+    private final Frame _outputPreds;
+
+    public PredictScoreResult(ModelMetrics.MetricBuilder<?> mb, Frame rawPreds, Frame outputPreds) {
+      _mb = mb;
+      _rawPreds = rawPreds;
+      _outputPreds = outputPreds;
+    }
+    
+    public final Frame getPredictions() {
+      return _outputPreds;
+    }
+
+    public ModelMetrics.MetricBuilder<?> getMetricBuilder(Frame adaptFrm) {
```

Comment:
`Frame adaptFrm` is unused. Is it a remnant of the previous version (IIRC it was used in `StackedEnsemblePredictScoreResult`)?

Reply:
good catch, I need to remove this argument as well